client/me/concierge/book/calendar-step.js

```diff
@@ -49,13 +49,13 @@ class CalendarStep extends Component {
 		this.props.recordTracksEvent( 'calypso_concierge_book_calendar_step' );
 	}
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillUpdate( nextProps ) {
-		if ( nextProps.signupForm.status === CONCIERGE_STATUS_BOOKED ) {
-			// go to confirmation page if booking was successfull
+	componentDidUpdate() {
+		if ( this.props.signupForm.status === CONCIERGE_STATUS_BOOKED ) {
+			// go to confirmation page if booking was successful
 			this.props.onComplete();
-		} else if ( nextProps.signupForm.status === CONCIERGE_STATUS_BOOKING_ERROR ) {
+		} else if ( this.props.signupForm.status === CONCIERGE_STATUS_BOOKING_ERROR ) {
 			// request new available times
+			alert( 'error' );
```

Comment:
Noting that `alert()` may and sometimes will cause unexpected effects, due to the fact that it prevents the user from accessing the rest of the UI until it is closed. Because of that, I personally tend to avoid testing/debugging with it. 

The [console](https://developer.mozilla.org/en-US/docs/Web/API/console) API is really useful in that case, especially `console.log()`, `console.warn()` and `console.error()`. 

We also tend to use the `debug` package widespread throughout Calypso. What's good about it is that you can have it in production code (like we often have in Calypso), but it will do nothing in production, and at the same time, in development, you can see debug messages in your console. You can read more about how we use it in Calypso [right here](https://github.com/Automattic/wp-calypso/blob/f8e2021db2b87bbf3cb1140e8adb72eaeac11efb/docs/development-workflow.md#debugging).

Reply:
Ah, I can't believe I forgot to remove it! 🤦‍♀️ I usually test with `console.log()` but sometimes use `alert()` to quickly see if I'm in the place I expect to be. But that makes sense to avoid using it for testing/debugging, I'll do that in the future. 👍 It's been removed now. And thanks for sharing info about `debug` in Calypso, I'll look into that!